predicators/main.py

```diff
@@ -309,7 +317,9 @@ def _generate_interaction_results(
 
 
 def _run_testing(env: BaseEnv, cogman: CogMan) -> Metrics:
-    test_tasks = env.get_test_tasks()
+    test_tasks = [
```

Comment:
Note: 
During training, we used the perceiver to turn all the environment tasks into tasks before passing the into the approach. 

But at test time, the "turning into a task" is done inside CogMan, where it's a bit more annoying to replace the goal with the alternative goal -- so I do it here, before we pass things into CogMan.

Reply:
interesting - we should probably fix that in CogMan later

Reply:
I don't know if there's anything to fix -- not sure. 

Currently, we only have CogMan for the test tasks, not for the train tasks. That seems to be by design, because CogMan is meant to manage episodes of environment interaction? 

I don't know if it makes sense to have the perceiver be external to the agent's cognitive manager for the train tasks but not for the test tasks. I feel like the agent is a black box and the input to it should be the environment tasks in both the train and test scenarios, esp if we're practically using the same perceiver to turn the environment tasks into tasks in both cases. 

Maybe it would make more sense if we used a different perceiver to create the train tasks, than the perceiver the agent was gonna use at test time.